app/account_linking.py

```diff
@@ -0,0 +1,256 @@
+from abc import ABC, abstractmethod
+from arrow import Arrow
+from dataclasses import dataclass
+from enum import Enum
+from typing import Optional
+
+from app.db import Session
+from app.models import PartnerSubscription, Partner, PartnerUser, User
+from app.utils import random_string
+
+
+class SLPlanType(Enum):
+    Free = 1
+    Premium = 2
+
+
+@dataclass
+class SLPlan:
+    type: SLPlanType
+    expiration: Optional[Arrow]
+
+
+@dataclass
+class PartnerLinkRequest:
+    name: str
+    email: str
+    partner_user_id: str
+    plan: SLPlan
+
+
+@dataclass
+class LinkResult:
+    user: User
+    strategy: str
+
+
+class LinkException(Exception):
```

Comment:
I think all exceptions are centralized in errors.py

Reply:
Moved exceptions to `errors.py`